### PR TITLE
[MINI PROV] : Fixed deployment logger bug

### PR DESCRIPTION
**Pull request by pranavPawar07**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered docs/notes.md](https://github.com/Seagate/cortx-s3server/blob/br/pranav/bug-fix/docs/notes.md)
[View rendered s3backgrounddelete/readme.md](https://github.com/Seagate/cortx-s3server/blob/br/pranav/bug-fix/s3backgrounddelete/readme.md)

### CODE REVIEW

File: scripts/provisioning/s3_setup [hunk 1]
```diff
@@ -30,11 +30,14 @@ from s3confstore.cortx_s3_confstore import S3CortxConfStore
 
 # Get the machine id from confstore
 s3_confkeys_store = S3CortxConfStore(f'yaml:///opt/seagate/cortx/s3/mini-prov/s3_prov_config.yaml', 'machine_id_index')
+s3_provkeys_store = S3CortxConfStore(f'yaml:///opt/seagate/cortx/s3/conf/s3.config.tmpl.1-node', 'config_log_index')
```
Comment (nileshgovande) [inline scripts/provisioning/s3_setup:33]: No hardcoading to s3.config.tmpl.1-node.
The config file will be passed to each of the phase

File: scripts/provisioning/s3_setup [hunk 2]
```diff
@@ -178,14 +173,22 @@ def create_logger():
...
-    _logger_directory = os.path.join(s3deployment_log_directory)
-    if not os.path.isdir(_logger_directory):
+def create_logger_directory(config_file):
+    """Create log directory if not exists."""
+    # Get the machine id from confstore
+    s3_confkeys_store = S3CortxConfStore(f'yaml:///opt/seagate/cortx/s3/mini-prov/s3_prov_config.yaml', 'machine_id_index')
+    s3_provkeys_store = S3CortxConfStore(f'yaml://{config_file}', 'config_log_index')
+
+    s3_prov_config_path = s3_provkeys_store.get_config('cortx>common>storage>log')
+    global s3deployment_log_file
```
Comment (nileshgovande) [inline scripts/provisioning/s3_setup:183]: Do we really need this 'Global' ?

Comment (vimalraghwani20) [reply scripts/provisioning/s3_setup:183]: image

Comment (pranavPawar07) [reply scripts/provisioning/s3_setup:183]: yes. somehow the s3_deployment_file value is not getting updated on create_logger function.

**Review by vimalraghwani20 — APPROVED** [review]

**Comment by vimalraghwani20** [comment]
Mini Provisioner job: 
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/CentOS-7.9.2009/job/S3-Server/343/console

**Pull request merged**